src/RunTests/RunVSTestTask.cs

```diff
@@ -0,0 +1,434 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+//
+// Licensed under the MIT license.
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+
+namespace Microsoft.Build
+{
+    /// <summary>
+    /// Runs tests with vstest.
+    /// </summary>
+    public class RunVSTestTask : Task
```

Comment:
Since this is launching an executable, inheriting from `ToolTask` may let you rely on common implementation for stuff like reading from stdin/out and other process-launch minutia and simplify the new code here.

Reply:
@nohwnd can you give some drive here as you worked on the tool task some time ago?

Reply:
updated to be a tooltask now